daq/host.py

```diff
@@ -570,17 +634,31 @@ def _host_dir_path(self):
     def _host_tmp_path(self):
         return os.path.join(self._host_dir_path(), 'tmp')
 
+    def _exists_new_ip_assignment(self, timestamp):
+        test_start = gcp.parse_timestamp(timestamp)
+        b_fn = lambda ip: gcp.parse_timestamp(ip["timestamp"]) < test_start
+        a_fn = lambda ip: gcp.parse_timestamp(ip["timestamp"]) >= test_start
+        before, after = list(filter(b_fn, self.all_ips)), list(filter(a_fn, self.all_ips))
+        return len(after) > 0 and after[-1]["ip"] != before[-1]["ip"]
+
+
     def _docker_callback(self, return_code=None, exception=None):
         self.timeout_handler = None # cancel timeout handling
         host_name = self._host_name()
         LOGGER.info('Host callback %s/%s was %s with %s',
                     self.test_name, host_name, return_code, exception)
         self._monitor_cleanup()
         failed = return_code or exception
-        if failed and self._fail_hook:
-            fail_file = self._FAIL_BASE_FORMAT % host_name
-            LOGGER.warning('Executing fail_hook: %s %s', self._fail_hook, fail_file)
-            os.system('%s %s 2>&1 > %s.out' % (self._fail_hook, fail_file, fail_file))
+        if failed:
+            if self._fail_hook:
+                fail_file = self._FAIL_BASE_FORMAT % host_name
+                LOGGER.warning('Executing fail_hook: %s %s', self._fail_hook, fail_file)
+                os.system('%s %s 2>&1 > %s.out' % (self._fail_hook, fail_file, fail_file))
+            # Check if new ip was assigned during test run and retry
```

Comment:
this seems non-deterministic... how does this work, then? It's just letting the change happen during some arbitrary test, and then restarting?  I'm not sold on that's the right thing.... seems like we might want it to be a more explicit phase...